docs/high-availability.md

```diff
@@ -0,0 +1,126 @@
+
+If you are running on physical machines, you can also use network attached redundant storage using an iSCSI or NFS interface.
+Alternatively, you can run a clustered file system like Gluster or Ceph.  Finally, you can also run a RAID array on each physical machine.
+
+Regardless of how you choose to implement it, you should make sure that your reliable storage is mounted to each machine.  If your storage
+is shared between the three masters in your cluster, you should create a different directory on the storage for each node.
+
+## Reliable nodes
+On each master, we are going to run a number of processes that implement the Kubernetes API.  The first step in making these reliable is
+to make sure that each automatically restarts when they fail.  To achieve this, we need to install a process watcher.  We choose to use
+the ```kubelet``` that we use on each of the nodes.  This is convenient, since we can use containers to distribute our binaries, we can
+establish resource limits, and introspect the resource usage of each daemon.  Of course, we also need something to monitor the kubelet
+itself (insert who watches the watcher jokes here).  To do this, we'll use the ```monit``` tool, but there are a number of alternate
+choices.  We also need to health check and restart the docker daemon, we'll use ```monit``` for this as well.
+
+You should install the [kubelet binary](https://storage.googleapis.com/kubernetes-release/release/v0.18.0/bin/linux/amd64/kubelet), the
+[/etc/init.d/kubelet](high/availability/init-kubelet) and [/etc/default/kubelet](high-availability/default-kubelet)
+scripts.  You should also install the monit daemon (```apt-get install monit```) and the [/etc/monit/conf.d/kubelet](high-availability/monit-kubelet) and
+[/etc/monit/conf.d/docker](high-availability/monit-docker) configs.
+
+## Clustered etcd
+Ok, you now have reliable storage, and reliable nodes, now it's time to start building a cluster.
+
+The full details of clustering etcd are beyond the scope of this document, lots of details are given on the
+[etcd clustering page](https://github.com/coreos/etcd/blob/master/Documentation/clustering.md).
+
+For simplicity, we'll use etcd's built in discovery to build our cluster.
+
+First, hit the etcd discovery service to create a new toke:
+
+```sh
+curl https://discovery.etcd.io/new?size=3
+```
+
+On each node, copy the [etcd.manifest](high-availability/etcd.manifest) file into ```/etc/kubernetes/manifests/etcd.manifest```
+
+Note that you should substitute the token URL you got above for ```${DISCOVERY_TOKEN}``` on all three machines,
+and you should substitute a different name (e.g. ```node-1```) for ${NODE_NAME} and the correct IP address
+for ```${NODE_IP}``` on each machine.
+
+Once you copy this into all three nodes, you should have a clustered etcd set up.  You can validate with
+```etcdctl member list``` and ```etcdctl cluster-health```.
```

Comment:
Put the text on separate lines from the three backticks, otherwise it confuses the Github PR review colorizer.